os/drivers/serial/serial.c

```diff
@@ -1214,6 +1235,18 @@ static int uart_open(FAR struct file *filep)
 	uint8_t tmp;
 	int ret;
 
+#ifdef CONFIG_PM
+	/* Register PM_UART_DOMAIN to access PM APIs during UART operations. */
+	if (pm_uart_domain_id == -1) {
+		ret = pm_domain_register(PM_UART_DOMAIN);
+		if (ret < 0) {
+			return ret;
+		} else {
+			pm_uart_domain_id = ret;
+		}
```

Comment:
```suggestion
pm_uart_domain_id = ret;
```

it has return, we don't need to eles